tests/rptest/clients/kubectl.py

```diff
@@ -284,19 +303,27 @@ def exec(self, remote_cmd, pod_name=None) -> str:
 
         self._install()
         if pod_name is None:
-            pod_name = f'rp-{self._cluster_id}-0'
+            if self._provider == 'azure':
+                pod_name = 'redpanda-broker-0'
```

Comment:
several of these differences are related to using operator v2, right?


Might it be better to gate these all on say `self.is_operator_v2()` and then that could just be implemented by checking `provider == "azure"` today or some other way in the future if we cut more things over to v2?